### ISSUE TITLE
Fix focus management and accessibility when toggling Hamburger menu button

### ISSUE THREAD

**Issue by majornista** [adobe/react-spectrum#2370]
# 🙋 Documentation Request

I noticed a couple small accessibility problems with Hamburger menu in documentation site caused by an improper CSS selector in the code.

1. The querySelector here selects the container element for the Hamburger menu button: https://github.com/adobe/react-spectrum/blob/11596864232ede67642f304e175a678eafcbb176/packages/dev/docs/src/client.js#L95
2. `aria-pressed` should be added to the button within the container element rather than the container element itself, and we should use the `setIsPressed` method to update the state rather than DOM manipulation:
https://github.com/adobe/react-spectrum/blob/11596864232ede67642f304e175a678eafcbb176/packages/dev/docs/src/client.js#L100
3. We should focus the Hamburger menu button rather than the container element here:  https://github.com/adobe/react-spectrum/blob/11596864232ede67642f304e175a678eafcbb176/packages/dev/docs/src/client.js#L103

I have a more robust fix for these issues using `useRef`.

## 🧢 Your Company/Team

Adobe/Accessibility
